cloudvolume/cloudvolume.py

```diff
@@ -158,22 +160,36 @@ def __init__(self, cloudpath, mip=0, bounded=True, autocrop=False, fill_missing=
     self.pid = os.getpid()
 
   @classmethod
-  def from_numpy(cls, arr, vol_path='file://tmp/'+generate_random_string(),
+  def from_numpy(cls, arr, vol_path='file:///tmp/image/'+generate_random_string(),
                   resolution=(4,4,40), voxel_offset=(0,0,0), 
-                  chunk_size=(64,64,64)):
-    mkdir(vol_path)
-    if arr.dtype == np.uint8 or arr.dtype == np.float32:
-      layer_type = 'image'
-    elif arr.dtype == np.uint32 or arr.dtype == np.uint64:
-      layer_type = 'segmentation'
-
-    # cloud volume use fortran order, so we need to transpose 
-    # arr = np.transpose(arr)
-    # import pdb; pdb.set_trace()
-    info = cls.create_new_info(1, layer_type, arr.dtype.name, 'raw', resolution, voxel_offset, arr.shape)
-    vol = CloudVolume(vol_path, info=info, bounded=True, autocrop=True) 
+                  chunk_size=(128,128,64), layer_type=None):
+    if re.match('^file://', vol_path):
```

Comment:
It might be more future proof to do something like:

```python
path = lib.extract_path(vol_path)
if path.protocol == 'file':
    mkdir(vol_path)
```

Reply:
modified as suggested.